docs/data/tree-view/items/items.md

```diff
@@ -0,0 +1,187 @@
+
+### Item identifier on `TreeView`
+
+You must pass a `nodeId` prop to each `TreeItem` component:
+
+```tsx
+<TreeView>
+  <TreeItem nodeId="tree-view-community" {...otherItemProps} />
+</TreeView>
+```
+
+### Item identifier on `RichTreeView`
+
+By default, the `RichTreeView` component looks for a property named `id` in the data set to get that identifier:
+
+```tsx
+const ITEMS = [{ id: 'tree-view-community' }];
+
+<RichTreeView items={ITEMS} />;
+```
+
+If the item's identifier is not called `id`, then you need to use the `getItemId` prop to tell the `RichTreeView` component where it's located:
+
+The following demo shows how to use `getItemId` to grab the unique identifier from a property named `internalId`:
+
+```tsx
+const ITEMS = [{ internalId: 'tree-view-community' }];
+
+function getItemId(item) {
+  return item.internalId;
+}
+
+<RichTreeView items={ITEMS} getItemId={getItemId} />;
+```
+
+{{"demo": "GetItemId.js", "defaultCodeOpen": false}}
+
+:::warning
+Just like the `items` prop, the `getItemId` function should keep the same JavaScript reference between two renders.
+Otherwise, the Tree View will re-generate its entire structure.
+
+It could be achieved by either defining the prop outside the component scope or by memoizing using the `React.useCallback` hook if the function reuses something from the component scope.
+:::
+
+## Item label
+
+Each item must have a label which does not need to be unique.
+
+### Item label on `TreeView`
+
+You must pass a `label` prop to each `TreeItem` component:
+
+```tsx
+<TreeView>
+  <TreeItem label="@mui/x-tree-view" {...otherItemProps} />
+</TreeView>
+```
+
+### Item label on `RichTreeView`
+
+By default, the `RichTreeView` component looks for a property named `label` in the data set to get that label:
+
+```tsx
+const ITEMS = [{ label: '@mui/x-tree-view' }];
+
+<RichTreeView items={ITEMS} />;
+```
+
+If the item's label is not called `label`, then you need to use the `getItemLabel` prop to tell the `RichTreeView` component where it's located:
+
+The following demo shows how to use `getItemLabel` to grab the unique identifier from a property named `internalLabel`:
+
+```tsx
+const ITEMS = [{ internalLabel: '@mui/x-tree-view' }];
+
+function getItemLabel(item) {
+  return item.internalId;
+}
+
+<RichTreeView items={ITEMS} getItemLabel={getItemLabel} />;
+```
+
+{{"demo": "GetItemId.js", "defaultCodeOpen": false}}
+
+:::warning
+Just like the `items` prop, the `getItemLabel` function should keep the same JavaScript reference between two renders.
+Otherwise, the Tree View will re-generate its entire structure.
+
+It could be achieved by either defining the prop outside the component scope or by memoizing using the `React.useCallback` hook if the function reuses something from the component scope.
+:::
+
+:::warning
+Unlike the `TreeView` component, the `RichTreeView` component only supports string labels, you can't pass React nodes to it.
```

Comment:
In `TreeView` we are doing the following: 

```tsx
const firstLetter = (contentRef.current?.textContent ?? '').substring(0, 1).toLowerCase(),
```

This first letter is used to move the focus when pressing a letter / digit to the 1st item that start with this letter / digit.

But this requires having the DOM element of the item **BEFORE** deciding the label 1st letter.

I see several solutions here:

1. We keep the 1st letter registration inside the `TreeItem` component even for `RichTreeView`. Since it's only used when interacting with the component, it is not needed on mount (unlike `disabled` which causes visual flickering).

2. We add a new `getLabelFirstLetter` prop which receives the item and returns the first letter. The benefit here is too give more control to the dev, because the current approach might not always be what the user want (for example, we could have a small `Chip` before the actual label in the `label` prop that should be ignored on the keyboard navigation).

3. We only support `label: string` on `RichTreeView` and encourage users to use custom `TreeItem` to add the additional information outside of the label. It's what the `Autocomplete` component does, `getOptionLabel` returns a `string` and you need to use `renderOption` if you want to render something more complex.

4. Other solution?